src/System.Numerics.Experimental/Half.cs

```diff
@@ -353,5 +356,402 @@ public bool TryFormat(Span<char> destination, out int charsWritten, ReadOnlySpan
         {
             throw new NotImplementedException();
         }
+
+        // -----------------------Start of to-half conversions-------------------------
+
+        public static implicit operator Half(int i)
+        {
+            bool sign = i < 0;
+            Half h = (uint)(sign ? -i : i); // Math.Abs but doesn't throw exception, because we cast it to uint anyway
+            return sign ? new Half((ushort)(h.m_value | SignMask)) : h;
+        }
+
+        public static implicit operator Half(uint i)
+        {
+            if (i == 0)
+                return default;
+
+            int shiftDist = BitOperations.LeadingZeroCount(i) - 21;
+            if (shiftDist >= 0)
+                return new Half(false, (ushort)(0x18 - shiftDist), (ushort) (i << shiftDist));
+
+            shiftDist += 4;
+            uint sig = shiftDist < 0 ? ShiftRightJam(i, -shiftDist) : i << shiftDist;
+            return RoundPackToHalf(false, (short) (0x1C - shiftDist), (ushort) sig);
+        }
+
+        public static implicit operator Half(long l)
+        {
+            bool sign = l < 0;
+            Half h = (ulong)(sign ? -l : l); // Math.Abs but doesn't throw exception, because we cast it to ulong anyway
+            return sign ? new Half((ushort)(h.m_value | SignMask)) : h;
+        }
+
+        public static implicit operator Half(ulong l)
+        {
+            if (l == 0)
+                return default;
+
+            int shiftDist = BitOperations.LeadingZeroCount(l) - 53;
+
+            if (shiftDist >= 0)
+                return new Half(false, (ushort)(0x18 - shiftDist), (ushort)(l << shiftDist));
+
+            shiftDist += 4;
+            ushort sig = (ushort)(shiftDist < 0 ? ShiftRightJam(l, -shiftDist) : l << shiftDist);
+            return new Half(RoundPackToHalf(false, (short)(0x1C - shiftDist), sig));
+        }
+
+        public static implicit operator Half(short s)
+        {
+            return (int) s;
+        }
+
+        public static implicit operator Half(ushort s)
+        {
+            return (uint) s;
+        }
+
+        public static implicit operator Half(byte b)
+        {
+            return (uint) b;
+        }
+
+        public static implicit operator Half(sbyte b)
+        {
+            return (int) b;
+        }
+
+        private const uint SingleSignMask = 0x80000000;
```

Comment:
(thinking out loud) Maybe these should be put in a helper class since they will also be needed for `Binary128` and potentially others...

It might also just be worth exposing them on the actual types, but that needs an API proposal

Reply:
Some methods in the Utility region as well, I guess; the original implementation seemed to share parts of them (e.g. Create*NaN). Definitely worth moving them out into a helper class.

Reply:
Moved them to `Ieee754Helpers` class with some other methods (and potentially useful ones?)